test/spec/functions/merge.js

```diff
@@ -171,3 +171,9 @@ it('should not convert strings to arrays when merging arrays of `source`', funct
   expect(actual).toEqual({a: ['x', 'y', 'z']});
 });
 
+it('should not consider null as an object', function() {
+  expect(merge(null, {})).toEqual({});
+  expect(merge({this: 'b'}, {this: 'k'}, {this: null})).toEqual({this: null});
+  expect(merge({this: 'b'}, {this: null}, {this: undefined})).toEqual({this: null});
+  expect(merge({this: 'b'}, {this: null}, {this: 'k'})).toEqual({this: 'k'});
```

Comment:
I think it might be relevant to add a test with `new RegExp()` or `new Date()`, to check the this spec trait: `- treats non-plain objects as plain`
If this merge put in a place such as `functions/` it might end up being reused in theses situations.